docs/source/user_guide/automl.ipynb

```diff
@@ -645,17 +1598,1393 @@
+      "\n",
+      "*****************************\n",
+      "* Evaluating Batch Number 1 *\n",
+      "*****************************\n",
+      "\n",
+      "Elastic Net Classifier w/ Imputer + Standard Scaler:\n",
+      "\tStarting cross validation\n",
+      "\tFinished cross validation - mean Log Loss Binary: 0.077\n",
+      "Logistic Regression Classifier w/ Imputer + Standard Scaler:\n",
+      "\tStarting cross validation\n",
+      "\tFinished cross validation - mean Log Loss Binary: 0.077\n",
+      "\tHigh coefficient of variation (cv >= 0.2) within cross validation scores.\n",
+      "\tLogistic Regression Classifier w/ Imputer + Standard Scaler may not perform as estimated on unseen data.\n",
+      "\n",
+      "Search finished after 00:03            \n",
+      "Best pipeline: Logistic Regression Classifier w/ Imputer + Standard Scaler\n",
+      "Best pipeline Log Loss Binary: 0.076807\n"
+     ]
+    }
+   ],
+   "source": [
+    "from concurrent.futures import ThreadPoolExecutor\n",
+    "\n",
+    "from evalml.automl.engine import CFEngine\n",
+    "from evalml.automl.engine.cf_engine import CFClient\n",
+    "\n",
+    "# Use thread-level paralellism\n",
+    "threaded_cf_engine = CFEngine(CFClient(ThreadPoolExecutor()))\n",
+    "\n",
+    "automl_cf_threaded = AutoMLSearch(X_train=X, y_train=y,\n",
+    "                      problem_type=\"binary\",\n",
+    "                      allowed_model_families=[ModelFamily.LINEAR_MODEL],\n",
+    "                      engine = threaded_cf_engine)\n",
+    "automl_cf_threaded.search(show_iteration_plot = False)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 28,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2021-07-27T18:56:20.147287Z",
+     "start_time": "2021-07-27T18:56:20.145197Z"
+    }
+   },
    "outputs": [],
    "source": [
-    "automl.results"
+    "# from concurrent.futures import ProcessPoolExecutor\n",
+    "\n",
+    "# # Repeat the process but using process-level parallelism\n",
+    "# process_cf_engine = CFEngine(CFClient(ProcessPoolExecutor()))\n",
+    "\n",
+    "# automl_cf_process = AutoMLSearch(X_train=X, y_train=y,\n",
+    "#                       problem_type=\"binary\",\n",
+    "#                       allowed_model_families=[ModelFamily.LINEAR_MODEL],\n",
+    "#                       engine = process_cf_engine)\n",
+    "# automl_cf_process.search(show_iteration_plot = False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Parallelism with Dask\n",
+    "\n",
+    "Thread or process level parallelism can be explicitly evoked for the DaskEngine (as well as the CFEngine).  The `processes` can be set to `True` and the number of processes set using `n_workers`.  If `processes` is set to `False`, then the resulting parallelism will be threaded and `n_workers` will represent the threads used.  Examples of both will follow."
```

Comment:
Another hypernit: Can we add tic marks around `DaskEngine` and `CFEngine` here? It was done previously and would be nice to keep the pattern
![image](https://user-images.githubusercontent.com/22552445/127228663-8c1492d2-e8d3-411a-a804-06a92469d5ef.png)